segmentstore/server/src/main/java/io/pravega/segmentstore/server/containers/MetadataStore.java

```diff
@@ -141,6 +143,10 @@ public void close() {
      * </ul>
      */
     CompletableFuture<Void> createSegment(String segmentName, SegmentType segmentType, Collection<AttributeUpdate> attributes, Duration timeout) {
+        if (segmentType.isTransient() && !NameUtils.isTransientSegment(segmentName)) {
+            return Futures.failedFuture(new IllegalStateException("Invalid name '" + segmentName + "' for SegmentType: Transient"));
```

Comment:
`IllegalArgumentException` is more appropriate since it's an argument that's wrong.

`IllegalStateException` should be thrown when a method invocation is inconsistent with the current state of the object.